Source/DafnyCore/AST/TopLevelDeclarations.cs

```diff
@@ -1104,11 +1105,23 @@ public class ModuleDefinition : RangeNode, IDeclarationOrUsage, IAttributeBearin
   }
 
   public IToken NameToken => tok;
-  public override IEnumerable<Node> Children => (Attributes != null ? new List<Node> { Attributes } : Enumerable.Empty<Node>()).Concat<Node>(TopLevelDecls).
+  public override IEnumerable<Node> Children => (Attributes != null ?
+      new List<Node> { Attributes } :
+      Enumerable.Empty<Node>()).Concat<Node>(TopLevelDecls).
     Concat(RefinementQId == null ? Enumerable.Empty<Node>() : new Node[] { RefinementQId });
 
-  public override IEnumerable<Node> PreResolveChildren => Includes.Concat<Node>(TopLevelDecls).Concat(
-    PrefixNamedModules.Select(tuple => tuple.Item2));
+  private IEnumerable<Node> preResolveTopLevelDecls;
```

Comment:
Resolving a modules implies to transfer all the PrefixNamedModules and putting them in nested modules under TopLevelDecls. However, the formatter deals with structural declarations only, not semantics, so it was messing up with the indentation of modules like `module Outer.A {}` because it would then believe that "module" belongs to "module A {}" which is _inside_ `module Outer` so would be wrongly indented by two spaces.

Reply:
Another tricky one!